widgetssdk/src/main/java/com/glia/widgets/core/secureconversations/domain/IsMessageCenterAvailableUseCase.kt

```diff
@@ -0,0 +1,38 @@
+package com.glia.widgets.core.secureconversations.domain
+
+import com.glia.androidsdk.Engagement
+import com.glia.androidsdk.GliaException
+import com.glia.androidsdk.RequestCallback
+import com.glia.widgets.core.queue.GliaQueueRepository
+import com.glia.widgets.helper.rx.Schedulers
+import io.reactivex.disposables.CompositeDisposable
+
+class IsMessageCenterAvailableUseCase(
+    private val queueId: String,
+    private val queueRepository: GliaQueueRepository,
+    private val schedulers: Schedulers
+) {
+    private val disposable = CompositeDisposable()
+
+    fun execute(callback: RequestCallback<Boolean>) {
+        disposable.add(
+            queueRepository.queues
+                .subscribeOn(schedulers.computationScheduler)
+                .observeOn(schedulers.mainScheduler)
+                .subscribe(
+                { queues ->
+                    val messagingQueues = queues
+                        .filter { queue -> queue.id == queueId }
+                        .filter { queue -> queue.state.medias.any { media -> media == Engagement.MediaType.MESSAGING } }
```

Comment:
It is a bit confusing to me because it is the async messaging, but should it also check queue status if it is `OPEN`

Reply:
Or maybe not exactly check if the queue is `OPEN` as it could be `UNSTUFFED` or `FULL` which are probably okay for async. But what if the queue was explicitly `CLOSED`

Reply:
Okay, added a check for `OPEN` state as per discussion in Slack. 68e6692bb79eedd3fcb32de3bcd853c6396430df